consensus/consensus.go

```diff
@@ -98,4 +99,13 @@ type PoW interface {
 
 	// Hashrate returns the current mining hashrate of a PoW consensus engine.
 	Hashrate() float64
+
+	// CalcDifficulty returns the difficulty that a new block should have when created at time
+	// given the parent block's time and difficulty.
+	CalcDifficulty(*params.ChainConfig, uint64, *types.Header) *big.Int
+
+	// AccumulateRewards credits the coinbase of the given block with the mining
+	// reward. The total reward consists of the static block reward and rewards for
+	// included uncles. The coinbase of each uncle block is also rewarded.
+	AccumulateRewards(*params.ChainConfig, *state.StateDB, *types.Header, []*types.Header)
```

Comment:
I think consensus.Engine.Finalize already contains the reward accumulation within. I'm not sure you need an external method for it.